seed/static/seed/js/controllers/building_detail_controller.js

```diff
@@ -51,6 +133,83 @@ angular.module('BE.seed.controller.building_detail', [])
         isopen: false
     };
 
+    $scope.time_type_change = function(time_type){
+        $scope.time_type = time_type;
+    }
+
+    $scope.active_change = function(active){
+        $scope.active = active;
+    }
+
+    $scope.loopback_change = function(loopback){
+        $scope.loopback = loopback;
+    }
+
+    $scope.save_gb_request_info = function(){
+        //TODO Front-end validation and URL request testing
+
+        // read user front-end input
+        var gb_url_input = angular.element(document.querySelector('#gb_url'))[0].value;
+        var gb_subscription_id = angular.element(document.querySelector('#gb_subscription_id'))[0].value;
+        var min_time_para = angular.element(document.querySelector('#min_para'))[0].value;
+        var max_time_para = angular.element(document.querySelector('#max_para'))[0].value;
+        var time_type = $scope.time_type;
+        var date_pattern = angular.element(document.querySelector('#date_pattern'))[0].value;
+        var active_flag = $scope.active;
+        var loopback_flag = $scope.loopback;
+        // read user front-end input ends
+
+        // empty input check
+        if(!(gb_url_input && min_time_para && max_time_para && time_type && active_flag && gb_subscription_id
+                && ($scope.gb_req_flag==='Y' || loopback_flag)
+                && (time_type==='timestamp' || date_pattern))){
+            alert('Please fill all the fields!');
+            return;
+        }
+        // empty check ends
+
+        // get canonical building ID from $scope
+        var canonical_building_id = $scope.building.canonical_building;
+        var building_snapshot_id = $scope.building.id;
+        // get building ID ends
+
+        // assemble save-to-DB data
+        var data = {};
+        data['url'] = gb_url_input;
+        data['subscription_id'] = gb_subscription_id;
+        data['min_date_parameter'] = min_time_para;
+        data['max_date_parameter'] = max_time_para;
+        data['building_id'] = canonical_building_id;
+        data['time_type'] = time_type;
+        if(time_type === 'date'){
+            data['date_pattern'] = date_pattern;
+        }else {
+            data['date_pattern'] = '';
+        }
+        data['active'] = active_flag;
+        data['loopback'] = loopback_flag;
+        // assemble save-to-DB data ends
+
+        // call building_services save_gb_request_info method
+        $scope.$emit('show_saving');
+        building_services.save_gb_request_info(data)
+            .then(function (data){
+                // resolve promise
+                audit_service.get_building_logs($scope.building.canonical_building)
+                    .then(function(data){
+                        $scope.audit_logs = data.audit_logs;
+                    });
+                $scope.$emit('finished_saving');
+                alert('Saving finished');
```

Comment:
convert this from a convert to a angular alert. 

preferably as a toast.

![screen shot 2016-02-09 at 10 14 19 am](https://cloud.githubusercontent.com/assets/1907354/12924214/381fe42e-cf16-11e5-94bf-42d0815b743b.png)